tests/e2e/test_e2e.py

```diff
@@ -77,7 +77,7 @@ def test_e2e(caplog):
         corpus_parser = Parser(
             structural=True, lingual=True, visual=True, pdf_path=pdf_path
         )
-        corpus_parser.apply(doc_preprocessor, parallelism=PARALLEL)
+        corpus_parser.apply(doc_preprocessor, parallelism=PARALLEL, count=max_docs)
```

Comment:
Note that because `doc_preprocessor` is a generator, it has no length attribute to use to calculate the progress bar. Thus, the user either needs to pass in `max_docs` as the count (recommended, since it doesn't materialize everything) or wrap the `doc_preprocessor`: `corpus_parser.apply(list(doc_preprocessor), parallelism=PARALLEL)`.

Reply:
there is no way to get this value from doc_preprocessor?